payjoin-io/src/ohttp_keys.rs

```diff
@@ -0,0 +1,89 @@
+use ohttp::KeyConfig;
+use tokio::task::spawn_blocking;
+use url::Url;
+
+use crate::error::{Error, InternalError};
+
+/// Returns the ohttp keys from a payjoin server.
+///
+/// `proxy_endpoint` - The proxy the user want to use when
+/// requesting the ohttp keys from a payjoin server.
+/// `pj_endpoint` - The payjoin server endpoint to fetch the ohttp
+/// keys from.
+pub async fn fetch(proxy_endpoint: Url, pj_endpoint: &Url) -> Result<KeyConfig, Error> {
+    let ohttp_keys_url = pj_endpoint.join("/ohttp-keys")?;
+    let res = spawn_blocking(move || {
+        let proxy = match ProxyServer::new(&proxy_endpoint) {
+            Ok(p) => p,
+            Err(e) => return Err(e),
+        };
+        proxy.get(ohttp_keys_url.as_str()).call().map_err(Error::from)
+    })
+    .await?
+    .map_err(Error::from)?;
+    if res.status() != 200 {
+        return Err(Error(InternalError::Ureq(ureq::Error::Status(res.status(), res))));
+    }
+    let mut body = Vec::new();
+    let _ = res.into_reader().read_to_end(&mut body)?;
+    KeyConfig::decode(&body).map_err(Error::from)
+}
+
+// Proxy server struct.
+// This is used to make requests to the payjoin server.
+struct ProxyServer {
+    agent: ureq::Agent,
+}
```

Comment:
- ureq::Agent represents an Http client, not a servers.
- Use conventional BIP 77 naming. s/payjoin server/payjoin directory/ s/Proxy server/Ohttp relay/